client/src/client.rs

```diff
@@ -646,13 +646,22 @@ impl Client {
     /// # Errors
     /// Fails if subscribing to websocket fails
     pub fn listen_for_events(
-        &mut self,
+        &self,
         event_filter: EventFilter,
     ) -> Result<impl Iterator<Item = Result<Event>>> {
-        EventIterator::new(
-            &format!("{}/{}", &self.torii_url, uri::SUBSCRIPTION),
+        events_api::EventIterator::new(self.events_handler(event_filter)?)
+    }
```

Comment:
I don't understand how can I pass my own WS API implementation inside this method...

Reply:
This method stays the same as before from user's prospective. Just like for `Client::get_status` or `Client::request`.

The main new thing is `Client::events_handler`, which can be used for custom WS API implementation. It is used here as well.